outputs/policyreport.go

```diff
@@ -0,0 +1,316 @@
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+)
+
+//for n namespace specific reports with their count
+type pr struct {
+	report *wgpolicy.PolicyReport
+	count  int
+}
+
+const (
+	clusterPolicyReportBaseName = "falco-cluster-policy-report-"
+	policyReportSource          = "Falco"
+	highpri                     = "high"
+)
+
+var (
+	failbound int
+	//count for cluster policy report
+	repcount int
+	//slice of policyreports and their counts(type pr)
+	polreports = make(map[string]*pr)
+	//for cluster policyreport
+	report *wgpolicy.ClusterPolicyReport = &wgpolicy.ClusterPolicyReport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterPolicyReportBaseName,
+		},
+		Summary: v1alpha2.PolicyReportSummary{
+			Fail: 0,
+			Warn: 0, //to-do
+		},
+	}
+)
+
+func NewPolicyReportClient(config *types.Configuration, stats *types.Statistics, promStats *types.PromStatistics, statsdClient, dogstatsdClient *statsd.Client) (*Client, error) {
+	clientConfig, err := rest.InClusterConfig()
+	if err != nil {
+		clientConfig, err = clientcmd.BuildConfigFromFlags("", config.PolicyReport.Kubeconfig)
+		if err != nil {
+			fmt.Printf("[ERROR] :unable to load kube config file: %v", err)
+		}
+	}
+	crdclient, err := crdClient.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	report.ObjectMeta.Name += uuid.NewString()[:8]
+	return &Client{
+		OutputType:      "PolicyReport",
+		Config:          config,
+		Stats:           stats,
+		PromStats:       promStats,
+		StatsdClient:    statsdClient,
+		DogstatsdClient: dogstatsdClient,
+		Crdclient:       crdclient,
+	}, nil
+}
+
+// CreateReport creates PolicyReport/ClusterPolicyReport Resource in Kubernetes
+func (c *Client) CreateReport(falcopayload types.FalcoPayload) {
+	failbound = c.Config.PolicyReport.FailThreshold
+	r, namespaceScoped := newResult(falcopayload)
+	if namespaceScoped != "" {
+		forPolicyReports(c, namespaceScoped, r)
+	} else {
+		forClusterPolicyReport(c, r)
+	}
+}
+
+//newResult creates a new entry for Reports
+func newResult(FalcoPayload types.FalcoPayload) (c *wgpolicy.PolicyReportResult, namespaceScoped string) {
+	namespaceScoped = "" // decision variable to increment for policyreport and clusterpolicyreport //to do //false for clusterpolicyreport
+	var m = make(map[string]string)
+	for index, element := range FalcoPayload.OutputFields {
+		if index == "ka.target.namespace" || index == "k8s.ns.name" {
+			namespaceScoped = fmt.Sprintf("%v", element) //true for policyreport
+		}
+		m[index] = fmt.Sprintf("%v", element)
+	}
+	var pri string //initial hardcoded priority bounds
+	if FalcoPayload.Priority > types.PriorityType(failbound) {
+		pri = highpri
+	} else if FalcoPayload.Priority < types.PriorityType(failbound) {
+		pri = "low"
+	} else {
+		pri = "medium"
+	}
+	return &wgpolicy.PolicyReportResult{
+		Policy:      FalcoPayload.Rule,
+		Category:    "SI - System and Information Integrity",
```

Comment:
Where this description comes from? Is there a convention?

Reply:
Yes @Issif I used the NIST 800-53 categories as that is a well known standard used by many enterprises 
https://kubernetes.slack.com/archives/C906A4GAF/p1630510669011800